homeworks/dobrikbartaeva/3_conditionals_loops/rock_paper_scissors.py

```diff
@@ -0,0 +1,66 @@
+# hogyan számoljuk a pontokat - fügvénnyel, mert mindig ez a rule:
+def determine_winner(player1, player2, rounds, player_inputs):
+    outcomes = {
+        ("paper", "rock"): "Player1",
+        ("scissor", "paper"): "Player1",
+        ("rock", "scissor"): "Player1",
+    }
+    
+    if (player1, player2) in outcomes:
```

Comment:
Ezt egy picit tudod egyszerűsíteni:

```
    if player1 == player2:
        print(f"Same choosen,let's repeat Round_{rounds}")
        return False #ezt majd később használom, hogy továbbléphet-e a ciklus vagy sem
    elif (player1, player2) in outcomes:
        player_inputs["Score1"] += 1
        print(f"Player1 won Round_{rounds}!")
    else: 
        player_inputs["Score2"] += 1 
        print(f"Player2 won Round_{rounds}!")
    
    return True 
```

A sorrendet csak azért cseréltem fel mert így logikusabb a következők miatt: Egy kimeneted van ami False-t ad vissza, így azt megvizsgáljuk először, és egyből returnöljük a False-t.
Ha bármelyik másik ág fut le, akkor nem return-öljük a True-t mindkettőben, elég ha kiosztjuk a pontokat, és a legvégén egyszer visszaadjuk a True-t

Reply:
Valóban lefut így, de az indentálást nem értem, hogy miért az if-en "kívül" van a return true.
Szóval ha ugyanazt írták, akkor visszaadja a false-t, és mivel itt van egy return, ezért már ki is ugrik a def-ből és az első visszakapott returnnel fut tovább a "fő" kód ami ezt a defet használja?

Reply:
Igen! 
Döntetlen esetén a return False-nál már tovább sem olvassa és egyből kilép a function-ből.
Az elif-ben és az else-ben pedig redundáns lenne ugyanazt a return True-t. Miután kilép bármelyik ágból (elif vagy else), akkor kilép True-val és kész.